src/ziggurat/messaging/producer.clj

```diff
@@ -116,16 +127,21 @@
         channel-queue-timeout-ms (get-channel-retry-queue-timeout-ms topic-entity channel)
         exponential-backoff-enabled (channel-retries-exponential-backoff-enabled topic-entity channel)]
     (if exponential-backoff-enabled
-      (let [retry-count (-> (ziggurat-config) :retry :count)
-            channel-retry-count (get-channel-retry-count topic-entity channel)
-            message-retry-count (:retry-count message-payload)]
-        (get-exponential-backoff (or channel-retry-count retry-count) message-retry-count (or channel-queue-timeout-ms queue-timeout-ms)))
+      (get-exponential-backoff-timeout-ms topic-entity channel message-payload (or channel-queue-timeout-ms queue-timeout-ms))
       (or channel-queue-timeout-ms queue-timeout-ms))))
 
-(defn publish-to-channel-delay-queue [channel message-payload]
-  (let [{:keys [exchange-name queue-timeout-ms]} (:delay (rabbitmq-config))
-        topic-entity  (:topic-entity message-payload)
+(defn get-delay-exchange-name [topic-entity channel message-payload]
+  (let [{:keys [exchange-name]} (:delay (rabbitmq-config))
         exchange-name (prefixed-channel-name topic-entity channel exchange-name)
+        exponential-backoff-enabled (channel-retries-exponential-backoff-enabled topic-entity channel)]
+    (if exponential-backoff-enabled
+      (let [exponential-backoff (get-exponential-backoff topic-entity channel message-payload)]
+        (str (name exchange-name) "_" exponential-backoff))
```

Comment:
in this case the first retry will have name `exchange_10` and it will go down to `exchange_1`. Can we have exchange names that express more intent? I do not want to couple retry counts with exchange names

Reply:
@theanirudhvyas first retry will go to exchange_1, get backoff exponent is `max_retry` - `message_retry_count`